banzai_nres/dbs.py

```diff
@@ -69,6 +69,34 @@ def diff_luminosity(cls, value):
         return func.abs(cls.luminosity - value)
 
 
+class Classification(Base):
+    __tablename__ = 'classifications'
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    ra = Column(Float)
+    dec = Column(Float)
+    T_effective = Column(Float)
+    log_g = Column(Float)
+    metallicity = Column(Float)
+    alpha = Column(Float)
+    Index('idx_radec', "ra", "dec")
+
+    @hybrid_method
+    def diff_ra(self, value):
+        return abs(self.ra - value)
+
+    @diff_ra.expression
+    def diff_ra(cls, value):
+        return func.abs(cls.ra - value)
```

Comment:
I think it would be better to do the matching using units of ra*cos(dec) rather than just ra -- this will deal with the distorted coordinate grid near the poles. I worry that otherwise the code might pick an incorrect matching record.

Reply:
I just did the correct great circle distance. That should take care of this and the next comment.